pkg/kubelet/kubelet_pods.go

```diff
@@ -1520,10 +1520,20 @@ func (kl *Kubelet) generateAPIPodStatus(pod *v1.Pod, podStatus *kubecontainer.Po
 			klog.V(4).InfoS("Cannot get host IPs", "err", err)
 		} else {
 			s.HostIP = hostIPs[0].String()
-			if kubecontainer.IsHostNetworkPod(pod) && s.PodIP == "" {
-				s.PodIP = hostIPs[0].String()
-				s.PodIPs = []v1.PodIP{{IP: s.PodIP}}
-				if len(hostIPs) == 2 {
+			// HostNetwork Pods inherit the NodeIPs as PodIPs.
+			// For historical reasons, these PodIPs are immutable once set,
+			// If the Node changes its IPs the user has to restart the Kubelet
+			// so the HostNetwork pods update their IPs. We may change this in a future
+			// https://github.com/kubernetes/enhancements/pull/3000
+			// but since it changes the current behavior "here be dragons"
```

Comment:
I'm not sure you need to explain that much. Also, you spend several sentences explaining that `podIPs` is immutable, and then add code that changes it.

```
// HostNetwork Pods inherit the node IPs as PodIPs. They are immutable once set,
// other than that if the node becomes dual-stack, we add the secondary IP.
```

(though probably we should remove the second IP on dual-stack-to-single-stack downgrade too)

Reply:
yeah, I see my gap on the language, I try to mean that once a PodIP exist that never changes, however, you can add a new IP if not exist, is not the array that is immutable (it can grow), is the element in the array that can't change once set

Reply:
> (though probably we should remove the second IP on dual-stack-to-single-stack downgrade too)

this can open a new set of corner cases and we don't have a good history here, per example, I think that all the dual stack services that are stored will fail in validation on a migration like this.